dom.bs

```diff
@@ -6770,6 +6770,22 @@ invoked, must run these steps:
  <li><p>If <a>context object</a> is a <a for=Element>shadow host</a>, then <a>throw</a> an
  "{{InvalidStateError!!exception}}" {{DOMException}}.
 
+ <li>
+  <p>If <a>context object</a>'s <a for=Element>local name</a> is a
+  <a>valid custom element name</a>, then:</p>
+
+  <ol>
+   <li><p>Let <var>definition</var> be the result of
+   <a lt="look up a custom element definition">looking up a custom element definition</a> given
+   <a>context object</a>'s <a for=Node>node document</a>, its <a for=Element>namespace</a>, its
+   <a for=Element>local name</a>, and null as <a><code>is</code> value</a>.</p></li>
+
+   <li><p>If <var>definition</var> is not null and <var>definition</var>'s
+   <a lt="concept-custom-element-definition-disable-shadow">disable shadow</a> is true,
```

Comment:
I'm not sure what's the correct way to make a link to "disable shadow" flag.

Reply:
Once HTML has exported this term it should work. I wonder if we should throw InvalidStateError instead as this would allow distinguishing between it being a shadow host and it being disabled, no?

Reply:
I thought that ```InvalidStateError``` was not suitable because the context object can't be 'valid' state again.  But I found the context object was valid before calling custom element constructor.  So ```InvalidStateError``` seems to make sense.

```attachInternals()``` has the same issue.

Reply:
Ah okay, let's change both then.

Reply:
BTW, the previous step checks element types, and this step is also a kind of element type check.  NotSupportedError is consistent with the previous step.

Reply:
But how would we address this issue then? Or you don't think it's an issue? I suppose we could also change the exception of it already having a shadow tree further down.

Reply:
I think either of NotSupportedError or InvalidStateError is ok.  I prefer NotSupportedError slightly.

Reply:
So we use NotSupportedError when there's already a shadow tree? I'd be okay with such a change.

Reply:
> So we use NotSupportedError when there's already a shadow tree? 

I don't think so.  It's definitely an issue of context object's state.  I think InvalidStateError is appropriate.

Reply:
You appear to be arguing for exposing the distinction between a shadow root being forbidden and one being used. I care about removing that distinction. I don't really care about types of exceptions as it's not clear to me we've been consistent enough for the differences between them to be meaningful. What do you propose?

Reply:
ok, I understand your intention.
Then I think NotSupportedError is ok for all of four DOMExceptions thrown by attachShadow(). IMO InvalidStateError is a specific subset of NotSupportedError.

Anyway, this PR doesn't change the DOMException type of the already-attached case.  It's not in the scope of this PR.

Reply:
If you want to do them separately I think we'll need to first change the existing exceptions then and then land this PR. Otherwise landing this would result in a broken state.

Reply:
I posted https://github.com/whatwg/dom/pull/761 and will now work on tests and browser bugs.